<launch>
  <arg name="ns" default="human1"/>
  <arg name="node_start_delay" default="4.0" />

  <!-- move_base node with hateb with namespace-->
  <node pkg="move_base" type="move_base" name="move_base" output="screen" launch-prefix="bash -c 'sleep $(arg node_start_delay); $0 $@' ">
    <remap from="/map" to="map_static"/>
    <remap from="odom" to="interface/in/human_odom"/>
    <remap from="cmd_vel" to="cmd_geo"/>
    <remap from="/move_base/HATebLocalPlannerROS/agents_states" to="/$(arg ns)/move_base/HATebLocalPlannerROS/agents_states" />
    <remap from="/move_base/GlobalPlanner/make_plan" to="/$(arg ns)/move_base/GlobalPlanner/make_plan"/>
    <remap from="/tracked_agents" to="/$(arg ns)/tracked_agents"/>

    <rosparam file="$(find inhus_navigation)/config/cohan_nav_config/move_base_params_human.yaml" command="load" />
    <rosparam file="$(find inhus_navigation)/config/cohan_nav_config/global_costmap_params_human.yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find inhus_navigation)/config/cohan_nav_config/local_costmap_params_human.yaml" command="load" ns="local_costmap" />
    <rosparam file="$(find inhus_navigation)/config/cohan_nav_config/hateb_local_planner_params_human.yaml" command="load" ns="HATebLocalPlannerROS" />

    <param name="ns" value="$(arg ns)"/>
    <param name="base_global_planner" value="global_planner/GlobalPlanner" />
    <param name="base_local_planner" value="hateb_local_planner/HATebLocalPlannerROS" />
    <!-- <param name="GlobalPlanner/allow_unknown" value="false" /> -->

    <param name="global_costmap/robot_base_frame" value="$(arg ns)/base_link"/>
    <param name="local_costmap/robot_base_frame" value="$(arg ns)/base_link"/>
    <param name="local_costmap/global_frame" value="$(arg ns)/odom"/>
  </node>

</launch>
